router.go

```diff
@@ -142,6 +142,10 @@ type Router struct {
 	// Custom OPTIONS handlers take priority over automatic replies.
 	HandleOPTIONS bool
 
+	// Configurable http.Handler which is called after modifying the automatic reply
+	// to OPTIONS requests. If HandleOptions is false this won't be used
+	OPTIONSPostHandler http.Handler
```

Comment:
That was part of my problem :) Perhaps instead of OPTIONSPostHandler I should have used OPTIONSAfterHandler. *shrugs*

I understand that the pattern is HandleX should call X, but where X = OPTIONS this doesn't work :(

How about OptionsHandler? The POST isn't really required given HandleMethodNotAllowed doesn't specify in the function name that it is inserting after the Allow header. I would then update the comments to be more inline with those for HandleMethodNotAllowed and MethodNotAllowed.

Thoughts?